src/Microsoft.ML.Transforms/CategoryImputer.cs

```diff
@@ -0,0 +1,1876 @@
+            }
+
+            protected override bool ReleaseHandle()
+            {
+                // Not sure what to do with error stuff here.  There shoudln't ever be one though.
+                return _destroySaveDataHandler(handle, _dataSize, out IntPtr errorHandle);
+            }
+        }
+
+        [DllImport("Featurizers", EntryPoint = "GetErrorInfoString", CallingConvention = CallingConvention.Cdecl), SuppressUnmanagedCodeSecurity]
+        private static extern bool GetErrorInfoString(IntPtr error, out IntPtr errorHandleString, out IntPtr errorHandleStringSize);
+
+        [DllImport("Featurizers", EntryPoint = "DestroyErrorInfoString", CallingConvention = CallingConvention.Cdecl), SuppressUnmanagedCodeSecurity]
+        private static extern bool DestroyErrorInfoString(IntPtr errorString, IntPtr errorStringSize);
+
+        [DllImport("Featurizers", EntryPoint = "DestroyErrorInfo", CallingConvention = CallingConvention.Cdecl), SuppressUnmanagedCodeSecurity]
+        private static extern bool DestroyErrorInfo(IntPtr error);
+        internal class ErrorInfoSafeHandler : SafeHandleZeroOrMinusOneIsInvalid
+        {
+            public ErrorInfoSafeHandler(IntPtr handle) : base(true)
+            {
+                SetHandle(handle);
+            }
+
+            protected override bool ReleaseHandle()
+            {
+                // Not sure what to do with error stuff here.  There shoudln't ever be one though.
+                return DestroyErrorInfo(handle);
+            }
+        }
+
+        internal class ErrorInfoStringSafeHandler : SafeHandleZeroOrMinusOneIsInvalid
+        {
+            private IntPtr _length;
+            public ErrorInfoStringSafeHandler(IntPtr handle, IntPtr length) : base(true)
+            {
+                SetHandle(handle);
+                _length = length;
+            }
+
+            protected override bool ReleaseHandle()
+            {
+                // Not sure what to do with error stuff here.  There shoudln't ever be one though.
+                return DestroyErrorInfoString(handle, _length);
+            }
+        }
+
+        #endregion
+
+        #region FitResult
+
+        internal enum FitResult : byte
+        {
+            Complete = 0, Continue, ResetAndContinue
+        }
+
+        #endregion
+
+        #region ColumnInfo
+
+        // REVIEW: Since we can't do overloading on the native side due to the C style exports,
+        // this was the best way I could think handle it to allow for any conversions needed based on the data type.
+
+        #region BaseClass
+
+        internal delegate bool DestroyCppTransformerEstimator(IntPtr estimator, out IntPtr errorHandle);
+        internal delegate bool DestroyTransformerSaveData(IntPtr buffer, IntPtr bufferSize, out IntPtr errorHandle);
+        internal delegate bool DestroyTransformedDataCpp(IntPtr output, IntPtr outputSize, out IntPtr errorHandle);
```

Comment:
Use Native instead of Cpp

Reply:
Done, changed all occurrences from `Cpp` to `Native`.